rasa/core/policies/rule_policy.py

```diff
@@ -214,85 +310,125 @@ def predict_action_probabilities(
-                # was applied inside the form.
-                # Rules might not explicitly switch back to the `Form`.
-                # Hence, we have to take care of that.
-                predicted_listen_from_general_rule = recalled is None or (
-                    domain.action_names[recalled] == ACTION_LISTEN_NAME
-                    and f"active_form_{active_form_name}" not in key
-                )
-                if predicted_listen_from_general_rule:
-                    logger.debug(f"Predicted form '{active_form_name}'.")
-                    result[domain.index_for_action(active_form_name)] = 1
-                    return result
-
-                # Since rule snippets inside the form contain only unhappy paths,
-                # notify the form that
-                # it was predicted after an answer to a different question and
-                # therefore it should not validate user input for requested slot
-                predicted_form_from_form_rule = (
-                    domain.action_names[recalled] == active_form_name
-                    and f"active_form_{active_form_name}" in key
-                )
-                if predicted_form_from_form_rule:
-                    logger.debug("Added `FormValidation(False)` event.")
-                    tracker.update(FormValidation(False))
-
-            if recalled is not None:
-                logger.debug(
-                    f"There is a rule for next action "
-                    f"'{domain.action_names[recalled]}'."
-                )
-
-                result[recalled] = 1
-            else:
-                logger.debug("There is no applicable rule.")
+        # there could be several negative rules
+        negative_recalled = [
+            self.lookup[NEGATIVE_RULES].get(key) for key in negative_keys
+        ]
 
-        return result
+        active_form_name = tracker.active_form_name()
+        if active_form_name:
+            # Check if a rule that predicted action_listen
+            # was applied inside the form.
+            # Rules might not explicitly switch back to the `Form`.
+            # Hence, we have to take care of that.
+            predicted_listen_from_general_rule = (
+                recalled == ACTION_LISTEN_NAME
+                and ACTIVE_FORM_PREFIX + active_form_name not in key
+            )
+            if predicted_listen_from_general_rule:
+                if NO_ACTIVE_FORM not in negative_recalled:
+                    # negative rules don't contain a key that corresponds to
+                    # the fact that active_form shouldn't be predicted
+                    logger.debug(
+                        f"Predicted form '{active_form_name}' by overwriting "
+                        f"'{ACTION_LISTEN_NAME}' predicted by general rule."
+                    )
+                    return active_form_name
+
+                # do not predict anything
+                recalled = None
```

Comment:
should this be this?
```suggestion
                recalled = self._default_predictions(domain)
```

Reply:
no recalled is the name of the action

Reply:
oh dear - I was a bit confused by jumping back and forth in the code